lib/paymentrequest.py

```diff
@@ -70,38 +70,35 @@ def load_ca_list():
 
 
 def get_payment_request(url):
-    u = urllib.parse.urlparse(url)
-    error = None
-    response = None
-    if u.scheme in ['http', 'https']:
-        try:
-            response = requests.request('GET', url, headers=REQUEST_HEADERS)
-            response.raise_for_status()
-            # Guard against `bitcoincash:`-URIs with invalid payment request URLs
-            if "Content-Type" not in response.headers \
-            or response.headers["Content-Type"] != "application/bitcoincash-paymentrequest":
-                data = None
-                error = "payment URL not pointing to a bitcoincash payment request handling server"
-            else:
-                data = response.content
-            print_error('fetched payment request', url, len(response.content))
-        except requests.exceptions.RequestException:
-            data = None
-            if response is not None:
-                error = response.content.decode()
-            else:
-                error = "payment URL not pointing to a valid server"
-    elif u.scheme == 'file':
-        try:
-            with open(u.path, 'r', encoding='utf-8') as f:
-                data = f.read()
-        except IOError:
-            data = None
-            error = "payment URL not pointing to a valid file"
+    data = error = None
+    try:
+        u = urllib.parse.urlparse(url)
+    except ValueError as e:
+        error = str(e)
     else:
-        raise BaseException("unknown scheme", url)
-    pr = PaymentRequest(data, error)
-    return pr
+        if u.scheme in ['http', 'https']:
+            try:
+                response = requests.request('GET', url, headers=REQUEST_HEADERS)
+                response.raise_for_status()
+                # Guard against `bitcoincash:`-URIs with invalid payment request URLs
+                if "Content-Type" not in response.headers \
+                or response.headers["Content-Type"] != "application/bitcoincash-paymentrequest":
+                    error = "payment URL not pointing to a bitcoincash payment request handling server"
+                else:
+                    data = response.content
+                print_error('fetched payment request', url, len(response.content))
+            except requests.exceptions.RequestException as e:
+                error = str(e)
+        elif u.scheme == 'file':
+            try:
+                with open(u.path, 'r', encoding='utf-8') as f:
+                    data = f.read()
+            except IOError:
+                error = "payment URL not pointing to a valid file"
+        else:
+            error = f"unknown scheme: '{u.scheme}'"
+
+    return PaymentRequest(data, error)
```

Comment:
The code now catches any errors which could be caused by a bad URL, and reports them through the return value.

Reply:
Ok. Did you check the codebase that all the callsites for this are ok with the API change? Or shall I do that?

Reply:
Hmm quick search shows it should be fine. Will check again though..

Reply:
I've had a look and I also think they'll be fine. It's not really an API change: the `error` field was already present. It just wasn't being consistently used for all network or URL errors.